tools/cbmc/proofs/DNS/DNSgetHostByName_a/DNSgetHostByName_a_harness.c

```diff
@@ -0,0 +1,62 @@
+/* FreeRTOS includes. */
+#include "FreeRTOS.h"
+#include "queue.h"
+#include "list.h"
+
+/* FreeRTOS+TCP includes. */
+#include "FreeRTOS_IP.h"
+#include "FreeRTOS_DNS.h"
+#include "FreeRTOS_IP_Private.h"
+
+
+/*This assumes that prvParseDNSReply, FreeRTOS_socket, FreeRTOS_GetUDPPayloadBuffer, pvPortMalloc and xTaskResumeAll
+are implemented correctly. The function func mimics the callback function.*/
+
+uint32_t prvParseDNSReply( uint8_t *pucUDPPayloadBuffer, size_t xBufferLength, TickType_t xIdentifier) { }
+
+Socket_t FreeRTOS_socket( BaseType_t xDomain, BaseType_t xType, BaseType_t xProtocol ){
+	Socket_t xCreatedSocket = malloc(sizeof(Socket_t));
+	return xCreatedSocket;
+}
+
+void * FreeRTOS_GetUDPPayloadBuffer(size_t xRequestedSizeBytes, TickType_t xBlockTimeTicks ) {
+	void *pvReturn = malloc(50);
```

Comment:
What is 50?

I wonder whether the returned buffer is supposed to be of length `xRequestedSizeBytes`, or some related number.

Otherwise, if 50 is a random number, please write a comment saying so. However, I would rather you find the RFC for UDP, and figure out if there is a maximum size for UDP payloads, and use that or something related (and write a comment with a link to the RFC).

If the number needs to be kept low for performance reasons, again: please write a comment saying so. Even then, I still think it would be worth modifying this so that it returns *up to* n bytes, where n could be zero.